### PR TITLE
UITOOL-209/get some results

### PR DESCRIPTION
**Pull request by aiheon**
### Proposed Changes

Mainly refactoring and the first draft for the results page.

**There's a long list of things not working and that's why the bar is now behind a feature flag**

- Second search don't work
- Arrow keys control of the dropdown is not working
- An Empty search should return ALL the results, not only 5
- The results page need a revamp (wainting on UX-UI mockup)
- Need to plug analytics on clicks to feed ML
- Dropdown sometime is stuck with query suggestions in it and cannot be closed

And there's probably more. Those are the most important to fix before launch.

Once this is merge, it will be easier to tackle task one pr at a time and split the remaining work.

 

### Potential Breaking Changes

None

### Acceptance Criteria

-   [ ] The proposed changes are covered by unit tests
-   [ ] The potential breaking changes are clearly identified
-   [ ] [README.md](https://github.com/coveo/plasma/blob/master/README.md) is adjusted to reflect the proposed changes (if relevant)

### CODE REVIEW

**Review by GermainBergeron — APPROVED** [review]

File: packages/website/src/pages/plasma-search/ResultPage.tsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+    return (
+        <>
+            <Section className="home flex-auto overflow-auto demo-content">
+                <Section className="section">
+                    <h2>Search Results:</h2>
+                    <div className="body-l-book plasma-description">Patate King!</div>
+                    <div className="tile-grid">
+                        {state.results.map((result: Result) =>
+                            result.title !== 'Vapor Design System' ? (
```
Comment (toofff) [inline packages/website/src/pages/plasma-search/ResultPage.tsx:51]: `Vapor Design System` is ok?

File: packages/website/src/search/StandaloneSearchBar.tsx [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+import classNames from 'classnames';
+import {EngineContext} from './engine/EngineContext';
+import {FeatureFlags} from '../FeatureFlags';
+
+interface SearchBarProps {
+    controller: StandaloneSearchBox;
+}
+const SearchBoxRerender: FunctionComponent<SearchBarProps> = (props) => {
+    const {controller} = props;
+    const [state, setState] = useState(controller.state);
```
Comment (toofff) [inline packages/website/src/search/StandaloneSearchBar.tsx:17]:
```suggestion
    const [state, setState] = useState<StandaloneSearchBoxState>(controller.state);
```

File: packages/website/src/search/StandaloneSearchBar.tsx [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+        return (
+            <>
+                {(state.suggestions.length > 0 || state.isLoadingSuggestions) && (
+                    <ListBox
+                        classes={['search-results-container']}
+                        isLoading={state.isLoadingSuggestions}
+                        items={results}
+                        onOptionClick={(item) => {
+                            controller.selectSuggestion(item.value);
+                        }}
```
Comment (toofff) [inline packages/website/src/search/StandaloneSearchBar.tsx:61]:
```suggestion
                        onOptionClick={(item) => controller.selectSuggestion(item.value)}
```

File: packages/website/src/search/StandaloneSearchBar.tsx [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+        );
+    };
+
+    return (
+        <div className="plasmaSearchBar">
+            <form
+                autoComplete="off"
+                onSubmit={(e) => {
+                    e.preventDefault();
+                }}
```
Comment (toofff) [inline packages/website/src/search/StandaloneSearchBar.tsx:74]:
```suggestion
                onSubmit={(event) => event.preventDefault()}
```

File: packages/website/src/search/componentTitleList.csv [hunk 5]
```diff
@@ -0,0 +1,98 @@
+Iconography
```
Comment (toofff) [inline packages/website/src/search/componentTitleList.csv:1]: Why this file?

File: packages/website/src/styles/plasmaSearchBar.scss [hunk 6]
```diff
@@ -3,7 +3,7 @@ $button-border-width: 1px;
 $search-bar-heigh: 48px;
 $search-bar-min-width: 648px;
 
-.standaloneSearchBar {
+.plasmaSearchBar {
```
Comment (toofff) [inline packages/website/src/styles/plasmaSearchBar.scss:6]: Use `kebab-case` for the CSS, please

File: packages/website/src/search/StandaloneSearchBar.tsx [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+        <Button classes={['search-button']} onClick={() => controller.submit()}>
+            <Svg svgName={'search'} className="icon mod-stroke" />
+        </Button>
+    );
+
+    const SuggestionListBox = () => {
+        const results: IItemBoxProps[] = state.suggestions.map((s) => ({
+            value: s.rawValue,
+            displayValue: s.highlightedValue,
+        }));
```
Comment (toofff) [inline packages/website/src/search/StandaloneSearchBar.tsx:50]: use variables longer than 3 characters, maintenance will thank you

File: packages/website/src/pages/plasma-search/ResultPage.tsx [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+
+    // const isNewSearchEvent = () => {
+    //     const currentSearchId = state.searchResponseId;
+
+    //     if (currentSearchId !== previousSearchId.current) {
+    //         previousSearchId.current = currentSearchId;
+    //         return true;
+    //     }
+    //     return false;
+    // };
```
Comment (toofff) [inline packages/website/src/pages/plasma-search/ResultPage.tsx:34]: Remove commented code please

File: packages/website/src/pages/plasma-search/ResultPage.tsx [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+import {
+    buildResultList,
+    ResultList as HeadlessResultList,
+    loadQueryActions,
+    Result,
+    SearchEngine,
+} from '@coveo/headless';
+import {Section, Tile} from '@coveord/plasma-react';
+import {useSearchParams} from 'react-router-dom';
+import {EngineContext} from '../../search/engine/EngineContext';
```
Comment (toofff) [inline packages/website/src/pages/plasma-search/ResultPage.tsx:13]:
```suggestion
import {
    buildResultList,
    ResultList as HeadlessResultList,
    loadQueryActions,
    Result,
    SearchEngine,
} from '@coveo/headless';
import {Section, Tile} from '@coveord/plasma-react';
import {FunctionComponent, useContext, useEffect, useRef, useState} from 'react';
import React from 'react';

import {useSearchParams} from 'react-router-dom';
import {EngineContext} from '../../search/engine/EngineContext';
```

File: packages/website/src/pages/plasma-search/ResultPage.tsx (hunk 1 above)
Comment (aiheon) [reply packages/website/src/pages/plasma-search/ResultPage.tsx:51]: No but i'm changing it in my current branch :see_no_evil: 
This one is more of a WIP

File: packages/website/src/search/componentTitleList.csv (hunk 5 above)
Comment (aiheon) [reply packages/website/src/search/componentTitleList.csv:1]: It's to feed ML for query suggestions :see_no_evil:

File: packages/website/src/pages/plasma-search/ResultPage.tsx (hunk 8 above)
Comment (aiheon) [reply packages/website/src/pages/plasma-search/ResultPage.tsx:34]: Not now :upside_down_face:

File: packages/website/src/search/StandaloneSearchBar.tsx (hunk 3 above)
Comment (aiheon) [reply packages/website/src/search/StandaloneSearchBar.tsx:61]: this is done in another branch

File: packages/website/src/search/StandaloneSearchBar.tsx (hunk 4 above)
Comment (aiheon) [reply packages/website/src/search/StandaloneSearchBar.tsx:74]: this is done in another branch

File: packages/website/src/search/componentTitleList.csv (hunk 5 above)
Comment (gdostie) [inline packages/website/src/search/componentTitleList.csv:1]: The list is outdated, some stuff was removed in during cleanup

**Review by gdostie — APPROVED** [review]

File: packages/website/src/search/componentTitleList.csv (hunk 5 above)
Comment (aiheon) [reply packages/website/src/search/componentTitleList.csv:1]: yeah I know :see_no_evil:  
I don't think I can remove candidates from the model but I can create another one :+1:

**Pull request merged**